components/kyma-environment-broker/internal/broker/logging.go

```diff
@@ -0,0 +1,48 @@
+package broker
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+)
+
+var openKeys = map[string]struct{}{
+	"sm_url":           {},
+	"xsappname":        {},
+	"globalaccount_id": {},
+	"subaccount_id":    {},
+}
+
+func hideSensitiveDataFromRawContext(d []byte) map[string]interface{} {
+	var data map[string]interface{}
+	err := json.Unmarshal(d, &data)
+	if err != nil {
+		return map[string]interface{}{}
+	}
+	return hideSensitiveDataFromContext(data)
+}
+
+func hideSensitiveDataFromContext(input map[string]interface{}) map[string]interface{} {
+	copy := input
+
+	for k, v := range copy {
+		if reflect.TypeOf(v).Kind() == reflect.String {
```

Comment:
nit: I think using type switch, in this case, is slightly more idiomatic go, [albeit with negligible performance benefits](https://www.reddit.com/r/golang/comments/riki8m/type_switching_vs_simple_reflection_performance/) otherwise